src/main/java/com/github/distributedtracing/ContinuationLocal.java

```diff
@@ -0,0 +1,170 @@
+package com.github.distributedtracing;
+
+public final class ContinuationLocal<T> {
+    private static InheritableThreadLocal<Object[]> localCtx =
```

Comment:
Is passive inheritance desired? Is it better to be explicit about passing context so that the mechanism by which context carries over thread boundaries is not dependent on how the thread was created. Or to put it another way, everyone should use executors or wrap runnables